rust/src/index/vector/pq.rs

```diff
@@ -279,18 +285,17 @@ impl ProductQuantizer {
             sub_vector_idx * num_centroids * sub_vector_width,
             num_centroids * sub_vector_width,
         );
-        // TODO: return reference instead of Arc?
-        Arc::new(as_primitive_array(&arr).clone())
+        Some(Arc::new(as_primitive_array(&arr).clone()))
     }
 
     /// Reconstruct a vector from its PQ code.
-    /// It only supports U8 for now.
+    /// It only supports U8 PQ code for now.
     pub fn reconstruct(&self, code: &[u8]) -> Arc<Float32Array> {
         assert_eq!(code.len(), self.num_sub_vectors);
         let mut builder = Float32Builder::with_capacity(self.dimension);
         let sub_vector_dim = self.dimension / self.num_sub_vectors;
         for i in 0..code.len() {
-            let centroids = self.centroids(i);
+            let centroids = self.centroids(i).unwrap();
```

Comment:
are these unwraps safe here? or should this return result?

Reply:
at the point where pq code is generated, the pq should have already been trained once? Guess we can return error to indicate this is a bug.